config/decoder.go

```diff
@@ -284,23 +232,7 @@ func (d *decoder) mapping(childKey string, value reflect.Value, def string) erro
 
 // Sets value to an object type.
 func (d *decoder) object(childKey string, value reflect.Value) error {
-	valueType := value.Type()
-	global := d.getGlobalProvider()
-
-	v2 := global.Get(childKey)
-
-	if !v2.HasValue() && valueType.Kind() == reflect.Ptr {
-		// in this case we will keep the pointer value as not defined.
-		return nil
-	}
-
-	if !value.CanSet() {
-		return nil
```

Comment:
We always can set value now, if it is not the case the unmarshall method did something really wrong.